### PR TITLE
Add GitHub Actions for Go

### LINKED ISSUE #161: Add GitHub Actions Integration

### PR DESCRIPTION
**Pull request by aod**
Close #161

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+      - run: go build -v .
+      - uses: actions/upload-artifact@v2
```
Comment (rexim) [inline .github/workflows/go.yaml:11]: Why are we uploading an artifact here? I thought `actions/upload-artifact` is used for sharing artifacts between the jobs and does not make much sense without `actions/download-artifact` in a different job. Am I missing something?

Comment (aod) [reply .github/workflows/go.yaml:11]: Copied this from my project and haven't really put much though behind it. Personally It's nice to have a executable which you can simply download via the Actions `details` page. So you don't have to compile it yourself, you can simply click on the build of a specific commit and download the program.

Comment (aod) [reply .github/workflows/go.yaml:11]: Oh and I thought you could reference artifacts within a GitHub release so this is kinda future proofing that I guess... Not sure if this is true though.

Comment (aod) [reply .github/workflows/go.yaml:11]: https://github.com/marketplace/actions/github-action-publish-binaries

Comment (aod) [reply .github/workflows/go.yaml]: After giving it some consideration I think this is an half-assed solution. I would rather take a different approach supporting not only Ubuntu, but other platforms as well. While using upload-artifact is an interesting solution for building and referencing artficats in releases. It should also not be part of the CI in every push and pull_request, only in actual releases. That was my intention after all.

**Comment by rexim** [comment]
@aod looks good to me! Thank you for your contribution! :+1:

**Pull request merged**